It seems that on 32-bit linux some of the ips are entered as `0x00000000` and thus being interpreted as a block end. And the check this PR added to `Profile.fetch()` is noticing.

See the `?`'s below

```julia
julia> Profile.clear()

julia> @profile busywait(1, 20)

julia> Profile.fetch()
ERROR: AssertionError: metadata stripping failed i=20 j=0 data[1:i]=UInt32[0xf7f39547, 0xf7f025cb, 0xf7af6a0c, 0xf79f9ae5, 0xe8694f8e, 0xe86c1831, 0xe8875ac0, 0xf799c4f7, 0xf79c4202, 0xf79f952c, 0xf7efb634, 0xf7e0c8f9, 0x00000004, 0xefb94010, 0x247e7c04, 0x00000002, 0x00000000, 0xf7f3a10c, 0xf7dca17e, 0xf7dca2e2]
Stacktrace:
 [1] fetch(; include_meta::Bool)
   @ Profile /buildworker/worker/package_linux32/build/usr/share/julia/stdlib/v1.8/Profile/src/Profile.jl:519
 [2] fetch()
   @ Profile /buildworker/worker/package_linux32/build/usr/share/julia/stdlib/v1.8/Profile/src/Profile.jl:494
 [3] top-level scope
   @ REPL[65]:1

julia> d = Profile.fetch(include_meta = true);

julia> for i in 1:length(d)-1
           if d[i + 1] == 0 && !in(d[i], [1,2]) 
               # the entry before a block end is the idle state, which can only be 1 or 2 (given we add 1 to the state val to avoid zero)
               # so indicate anything zeros that don't match that pattern
               @info i
           end
       end
[ Info: 9875
[ Info: 36298
[ Info: 38375
[ Info: 62623

julia> d[9850:9900]
51-element Vector{UInt32}:
 0xf79f952c
 0xf7efb634
 0xf7e0c8f9
 0x00000002 # threadid
 0xefb90010 # taskid
 0x3bf946fc # cycleclock
 0x00000002 # idle state (+1)
 0x00000000 # block end
 0xf7f39547
 0xf7f025cb
 0xf7af6a0c
 0xf79f9ae5
 0xe8694f8e
 0xe86c1831
 0xe8875ac0
 0xf799c4f7
 0xf79c4202
 0xf79f952c
 0xf7efb634
 0xf7e0c8f9
 0x00000004 # threadid
 0xefb94010 # taskid
 0x3bfa0ede # cycleclock
 0x00000002 # idle state (+1)
 0x00000000 # block end
 0xf7dca235
 0x00000000 # ???????
 0x75c085c0
 0x00000001 # threadid
 0xefb08010 # taskid
 0x3bfb1482 # cycleclock
 0x00000001 # idle state (+1)
 0x00000000 # block end
 0xf7f39547
 0xf7f025cb
 0xf7af6a0c
 0xf79f9ae5
 0xe8694f8e
 0xe86c1831
 0xe8875ac0
 0xf799c4f7
 0xf79c4202
 0xf79f952c
 0xf7efb634
 0xf7e0c8f9
 0x00000002 # threadid
 0xefb90010 # taskid
 0x3c30e878 # cycleclock
 0x00000002 # idle state (+1)
 0x00000000 # block end
 0xf7f3a10c
```

@vtjnash any idea what's happening here?

--- a/stdlib/Profile/src/Profile.jl
+++ b/stdlib/Profile/src/Profile.jl
@@ -500,17 +500,26 @@ function fetch(;include_meta = false)
     end
     data = Vector{UInt}(undef, len)
     GC.@preserve data unsafe_copyto!(pointer(data), get_data_pointer(), len)
-    if include_meta
+    if include_meta || isempty(data)
         return data
     else
-        nblocks = count(iszero, data)
+        nblocks = 0
+        for i = 2:length(data)
+            if data[i] == 0 && in(data[i - 1], [1,2])
+                # detect block ends and count them
+                # linux 32 has been seen to have rogue ips equal to 0 so also check for the previous entry looking like an idle
+                # state metadata entry which can only be 1 or 2
+                nblocks += 1
+            end
+        end
         nmeta = 4 # number of metadata fields (threadid, taskid, cpu_cycle_clock, thread_sleeping)
         data_stripped = Vector{UInt}(undef, length(data) - (nblocks * nmeta))
         j = length(data_stripped)
         i = length(data)
         while i > 0 && j > 0
             data_stripped[j] = data[i]
-            if data[i] == 0
+            if i > 1 && data[i] == 0 && in(data[i - 1], [1,2])
+                # detect block end (same approach as above)
                 i -= nmeta
             end
             i -= 1
@@ -540,7 +549,7 @@ function parse_flat(::Type{T}, data::Vector{UInt64}, lidict::Union{LineInfoDict,
     for i in startframe:-1:1
         startframe - 1 <= i <= startframe - 4 && continue # skip metadata (it's read ahead below)
         ip = data[i]
-        if ip == 0
+        if i > 1 && ip == 0 && in(data[i - 1], [1,2]) # check that the field next to the zero is the idle metadata entry
             # read metadata
             thread_sleeping = data[i - 1] - 1 # subtract 1 as state is incremented to avoid being equal to 0
             # cpu_cycle_clock = data[i - 2]
@@ -788,7 +797,7 @@ function tree!(root::StackFrameTree{T}, all::Vector{UInt64}, lidict::Union{LineI
     for i in startframe:-1:1
         startframe - 1 <= i <= startframe - 4 && continue # skip metadata (its read ahead below)
         ip = all[i]
-        if ip == 0
+        if i > 1 && ip == 0 && in(all[i - 1], [1,2]) # check that the field next to the zero is the idle metadata entry
             # read metadata
             thread_sleeping = all[i - 1] - 1 # subtract 1 as state is incremented to avoid being equal to 0
             # cpu_cycle_clock = all[i - 2]

